site/_docs/avatica_json_reference.md

```diff
@@ -0,0 +1,666 @@
+  - { name: "PrepareRequest" }
+  - { name: "SchemasRequest" }
+  - { name: "TableTypesRequest" }
+  - { name: "TablesRequest" }
+  - { name: "TypeInfoRequest" }
+miscellaneous:
+  - { name: "ConnectionProperties" }
+  - { name: "TypedValue" }
+  - { name: "Signature" }
+  - { name: "Frame" }
+  - { name: "StatementHandle" }
+  - { name: "DatabaseProperty" }
+  - { name: "ColumnMetaData" }
+  - { name: "AvaticaParameter" }
+  - { name: "AvaticaType" }
+  - { name: "Rep" }
+  - { name: "CursorFactory" }
+  - { name: "Style" }
+responses:
+  - { name: "ResultSetResponse" }
+  - { name: "ExecuteResponse" }
+  - { name: "PrepareResponse" }
+  - { name: "FetchResponse" }
+  - { name: "CreateStatementResponse" }
+  - { name: "CloseStatementResponse" }
+  - { name: "CloseConnectionResponse" }
+  - { name: "ConnectionSyncResponse" }
+  - { name: "DatabasePropertyResponse" }
+---
+
+<!--
+{% comment %}
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to you under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+As Avatica uses JSON to serialize messages sent over an HTTP transport,
+the RPC layer is agnostic of the language used by a client. While the Avatica
+server is written in Java, this enables clients to interact with the server
+using any language instead of being limited to Java.
+
+A specification of the JSON request and response objects are documented
+below. Programmatic bindings for these JSON objects are only available
+in Java, so non-Java clients presently must re-implement language
+specific objects on their own. Efforts to use [Protocol Buffers](https://developers.google.com/protocol-buffers/)
+instead are underway that will provide native objects in many languages.
+
+## Index
+
+### Requests
+<ul>
+  {% for item in page.requests %}<li><a href="#{{ item.name | downcase }}">{{ item.name }}</a></li>{% endfor %}
+</ul>
+
+### Responses
+<ul>
+  {% for item in page.responses %}<li><a href="#{{ item.name | downcase }}">{{ item.name }}</a></li>{% endfor %}
+</ul>
+
+### Miscellaneous
+<ul>
+  {% for item in page.miscellaneous %}<li><a href="#{{ item.name | downcase }}">{{ item.name }}</a></li>{% endfor %}
+</ul>
+
+
+## Requests
+
+The collection of all JSON objects accepted as requests to Avatica.
+
+### CatalogsRequest
+
+{% highlight json %}
+{}
```

Comment:
I think each of the requests has a type attribute, right?

I don't remember whether the responses have a type parameter also.